src/components/views/rooms/MessageComposerInput.js

```diff
@@ -285,22 +285,20 @@ export default class MessageComposerInput extends React.Component {
             case 'focus_composer':
                 editor.focus();
                 break;
-
-            // TODO change this so we insert a complete user alias
-
             case 'insert_displayname': {
-                contentState = Modifier.replaceText(
-                    contentState,
-                    this.state.editorState.getSelection(),
-                    `${payload.displayname}: `,
-                );
-                let editorState = EditorState.push(this.state.editorState, contentState, 'insert-characters');
-                editorState = EditorState.forceSelection(editorState, contentState.getSelectionAfter());
-                this.onEditorContentChanged(editorState);
-                editor.focus();
+                // Pretend that we've autocompleted this user because keeping two code
+                // paths for inserting a user pill is not fun
+                const selection = this.state.editorState.getSelection();
+                const member = this.props.room.getMember(payload.user_id);
+                const completion = member ? member.name : payload.user_id;
```

Comment:
This'll lose the behaviour of stripping the ' (IRC)' suffix - is that intentional?